### PR TITLE
Prevent a store from displaying categories that are not active

### PR DESCRIPTION
**Pull request by jeckyl**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Reuse of controls present in 1.6 and absent from 1.7 to prevent a store from displaying categories that are not active
| Fixes | #13468
| Type?         | bug fix
| Category?     | FO
| How to test?  | Attempt to access a category in another store (in multi-shop mode) and you will see an empty page, however as it is displayed it disturbs the SEO. Detailed steps in  #13468

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21877)

### CODE REVIEW

**Review by matks — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @jeckyl,

I found an exception following these steps:

1. Install PrestaShop.
2. Click on Shop Parameters > General.
3. Scroll down to Enable Multistore
4. Click on Advanced Parameters > Multistore.
5. Create a new store and define the categories to view in this shop.
image
6. Establish a URL for the second shop.
7. When you access a product of a category (Clothes) that should not be seen, the product is displayed. 
8. Try to access the category through the breadcrumb
9. see the error

image

Thanks!

**Comment by matks** [comment]
Hello, it seems this PR is now stale.

I close this PR, following our [review guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/) "Pull Requests may be closed after 30 days of inactivity following a request for modifications.". If however you wish to continue working on it, please ping us 😉  we can reopen.

**Comment by jeckyl** [comment]
> Hello, it seems this PR is now stale.
> 
> I close this PR, following our [review guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/) "Pull Requests may be closed after 30 days of inactivity following a request for modifications.". If however you wish to continue working on it, please ping us 😉 we can reopen.

So we consider that the problem is corrected?

**Comment by PululuK** [comment]
Hi @khouloudbelguith 
When you are in debug mode this error tells me something and it doesn't seem to come from this PR.
Can you test please with and without this PR?

Thank you

**Comment by matks** [comment]
> > Hello, it seems this PR is now stale.
> > I close this PR, following our [review guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/) "Pull Requests may be closed after 30 days of inactivity following a request for modifications.". If however you wish to continue working on it, please ping us 😉 we can reopen.
> 
> So we consider that the problem is corrected?

No, but we understand that contributing is an investment of time and energy and that it can fade away in time. Sometimes contributors start a PR, a discussion starts then the contributor goes away. Keeping the PR open is at this time is useless as it will not be updated anymore. This is why stale PRs are closed.

Are you motivated to continue working on this PR 😊  ? I can reopen it if you want to.

**Comment by jeckyl** [comment]
> Are you motivated to continue working on this PR 😊 ? I can reopen it if you want to.

@PululuK posted a question, let's wait and see what the answer will be, because his question is relevant.

**Comment by Robin-Fischer-PS** [comment]
Hi everyone,

I managed to reproduce the bug identified by @khouloudbelguith without the PR (with debug mode enabled), only on develop (1.7.8) : 

1. Create a category
2. Disable the category
3. In FO, try to access the category by URL : The exception is displayed.

image

So it's not due to the PR, but the PR makes this bug more visible to the user (accessible directly through breadcrumb link instead of access only by URL).

It seems it's a problem with the 404 pages not being displayed on FO (in 1.7.7 following these steps, I have a 404 page)

I think this PR can be re-opened and validated (provided there is no other bug), and the bug must be fixed too (not in this PR of course).

WDYT @matks @PululuK ?

**Comment by PululuK** [comment]
Thank you @Robin-Fischer-PS  ! 
So I will reopen this PR and we need to create an issue (I think someone had already done it and it has been fixed ... but I can no longer find the issue or the PR 😔).

**Review by PululuK — APPROVED** [review]

**Comment by Robin-Fischer-PS** [comment]
@PululuK I did not find an issue on this, I will create it this afternoon if I still cannot find it. Thanks :)

**Comment by Robin-Fischer-PS** [comment]
Update : The issue is created : https://github.com/PrestaShop/PrestaShop/issues/24143

If it's quickly fixed (a core dev is looking at it), I suggest we wait for the fix to be merged before testing this PR (so we can test it in "normal" conditions). However, if the fix for #24143 takes a long time to be done, we will test this PR first.

**Comment by Progi1984** [comment]
@sarahdib & @Robin-Fischer-PS #24143 has been fixed.

**Comment by khouloudbelguith** [comment]
Hi @Progi1984,

This PR is still blocked because the last issue is fixed in branch 1.7.8.x and not in the develop branch.
We need to merge 178x into develop to test this PR.

Thank you!

**Review by matthieu-rolland — APPROVED** [review]

**Comment by Progi1984** [comment]
@khouloudbelguith We merged 177x => 178x => develop yesterday.

**Comment by matks** [comment]
> Hi everyone,
> 
> I managed to reproduce the bug identified by @khouloudbelguith without the PR (with debug mode enabled), only on develop (1.7.8) :
> 
> 1. Create a category
> 2. Disable the category
> 3. In FO, try to access the category by URL : The exception is displayed.
> 
> image
> 
> So it's not due to the PR, but the PR makes this bug more visible to the user (accessible directly through breadcrumb link instead of access only by URL).
> 
> It seems it's a problem with the 404 pages not being displayed on FO (in 1.7.7 following these steps, I have a 404 page)
> 
> I think this PR can be re-opened and validated (provided there is no other bug), and the bug must be fixed too (not in this PR of course).
> 
> WDYT @matks @PululuK ?

Hello everyone,

If I understand correctly, we now wait for a last QA validation. QA team will pick this PR and test it in the context of `develop`.

As `develop` is now fixed regarding https://github.com/PrestaShop/PrestaShop/issues/24143 then the test of this PR should be free of errors, and then we can validate this PR 👍

**Comment by florine2623** [comment]
Hello @jeckyl ,

Your PR doesn't fix the initial bug in the issue.
When testing the PR, we found this bug #24815, it might be the cause. What do you think ?

In the breadcrumb, I still see the "Clothes" category.
Capture d’écran 2021-06-04 à 16 05 31


In `develop` the exception found by @khouloudbelguith and @Robin-Fischer-PS is fixed.
But on the PR, I encounter the same exception when following the steps in https://github.com/PrestaShop/PrestaShop/pull/21877#issuecomment-727861338
Capture d’écran 2021-06-04 à 16 06 59


Please check and feedback.
Thanks!

**Comment by matks** [comment]
Hello, it seems this PR is now stale.

Following our [review guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/) "Pull Requests may be closed after 30 days of inactivity following a request for modifications." I close this Pull Request.

If however you are still interested in working on it, please ping us :) we can reopen. 😄 

Thank you for your contribution

**Comment by jeckyl** [comment]
Merci @matks
et surtout merci à @florine2623 pour sa participation.

4 reviewers de cette PR n'ont pas pu amener le code jusqu'à sa validation.

Franchement, je ne comprends plus rien.

**Comment by superteixeira** [comment]
Hello,
Still don't have a release date for the fix for this bug?
Thanks

**Pull request closed without merging**
